### PR TITLE
Add MPT-7B-instruct HuggingFace Model

### PR DESCRIPTION
**Pull request by apsonawane**
PR to add https://huggingface.co/mosaicml/mpt-7b-instruct in torchbenchmark.
Running mpt-7b-instruct requires ```trust_remote_code``` to be passed

### CODE REVIEW

File: torchbenchmark/models/hf_MPT_7b_instruct/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+sentencepiece
+datasets
```
Comment (xuzhao9) [inline torchbenchmark/models/hf_MPT_7b_instruct/requirements.txt:2]: Looks like this file is not needed?

Comment (apsonawane) [reply torchbenchmark/models/hf_MPT_7b_instruct/requirements.txt:2]: Yes will remove this

**Comment by apsonawane** [comment]
Rebased with recent changes no merge conflicts

**Comment by msaroufim** [comment]
You're still getting an upstream error from a broken PR - you can try rebasing or something easier is

```

git checkout main
git pull
git checkout your_branch
git merge main
# fix merge conflicts
git push
```

**Comment by apsonawane** [comment]
Updated the branch and pushed

File: torchbenchmark/models/hf_MPT_7b_instruct/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+devices:
+  NVIDIA A100-SXM4-40GB:
+    eval_batch_size: 1
+eval_benchmark: false
+eval_deterministic: false
+eval_nograd: true
+not_implemented:
+- jit: true
```
Comment (xuzhao9) [inline torchbenchmark/models/hf_MPT_7b_instruct/metadata.yaml:8]: We have removed jit from the unit test cases, could you please remove the `not_implemented: - jit: true` part from the metadata file?

**Review by xuzhao9 — APPROVED** [review]
LGTM

File: torchbenchmark/models/hf_MPT_7b_instruct/metadata.yaml (hunk 2 above)
Comment (apsonawane) [reply torchbenchmark/models/hf_MPT_7b_instruct/metadata.yaml:8]: Yes

File: torchbenchmark/models/hf_MPT_7b_instruct/__init__.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from torchbenchmark.tasks import NLP
+from torchbenchmark.util.framework.huggingface.model_factory import HuggingFaceModel
+
+class Model(HuggingFaceModel):
+    task = NLP.LANGUAGE_MODELING
+    # https://huggingface.co/mosaicml/mpt-7b
+    DEFAULT_TRAIN_BSIZE = 4
+    DEFAULT_EVAL_BSIZE = 1
+
+    def __init__(self, test, device, jit=False, batch_size=None, extra_args=[]):
```
Comment (xuzhao9) [inline torchbenchmark/models/hf_MPT_7b_instruct/__init__.py:10]: Similarly, can you remove the `jit=False` argument and the `jit=jit` argument below? Sorry I missed this in my previous review.

File: torchbenchmark/models/hf_MPT_7b_instruct/__init__.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+from torchbenchmark.util.framework.huggingface.model_factory import HuggingFaceModel
+
+class Model(HuggingFaceModel):
+    task = NLP.LANGUAGE_MODELING
+    # https://huggingface.co/mosaicml/mpt-7b
+    DEFAULT_TRAIN_BSIZE = 4
+    DEFAULT_EVAL_BSIZE = 1
+
+    def __init__(self, test, device, batch_size=None, extra_args=[]):
+        super().__init__(name="hf_MPT_7b_instruct", test=test, device=device, jit=jit, batch_size=batch_size, extra_args=extra_args)
```
Comment (xuzhao9) [inline torchbenchmark/models/hf_MPT_7b_instruct/__init__.py:11]: Could you please also remove the `jit=jit` argument here?  @apsonawane

Comment (apsonawane) [reply torchbenchmark/models/hf_MPT_7b_instruct/__init__.py:11]: Oh my bad. Sorry I am missing these things

Comment (xuzhao9) [reply torchbenchmark/models/hf_MPT_7b_instruct/__init__.py]: No worries!

**Pull request closed without merging**